driver_test.go

```diff
@@ -1029,3 +1060,42 @@ func getPodmanDriver(t *testing.T, harness *dtestutil.DriverHarness) *Driver {
 	require.True(t, ok)
 	return driver
 }
+
+// helper to start, destroy and inspect a long running container
+func startDestroyInspect(t *testing.T, taskCfg TaskConfig, taskName string) iopodman.InspectContainerData {
+	if !tu.IsCI() {
+		t.Parallel()
+	}
+
+	task := &drivers.TaskConfig{
+		ID:        uuid.Generate(),
+		Name:      taskName,
+		AllocID:   uuid.Generate(),
+		Resources: createBasicResources(),
+	}
+	require.NoError(t, task.EncodeConcreteDriverConfig(&taskCfg))
+
+	d := podmanDriverHarness(t, nil)
+	cleanup := d.MkAllocDir(task, true)
+	defer cleanup()
+
+	containerName := BuildContainerName(task)
+
+	_, _, err := d.StartTask(task)
+	require.NoError(t, err)
+
+	defer d.DestroyTask(task.ID, true)
+
+	// Attempt to wait
+	waitCh, err := d.WaitTask(context.Background(), task.ID)
+	require.NoError(t, err)
+
+	select {
+	case <-waitCh:
+		t.Fatalf("wait channel should not have received an exit result")
+	case <-time.After(time.Duration(tu.TestMultiplier()*2) * time.Second):
```

Comment:
this seems like we will always wait for 2 seconds? maybe instead we can bring over or use `testutil.WaitForResult` https://github.com/hashicorp/nomad/blob/master/testutil/wait.go#L17